user-operator/src/main/java/io/strimzi/operator/user/model/acl/SimpleAclRule.java

```diff
@@ -133,13 +138,24 @@ public static SimpleAclRule fromAclBinding(AclBinding aclBinding) {
     }
 
     /**
-     * Creates SimpleAclRule object based on AclRule object which is received as part ofthe KafkaUser CRD.
+     * Creates SimpleAclRule object based on AclRule object which is received as part of the KafkaUser CRD.
      *
-     * @param rule  AclRule object from KafkaUser CR
+     * @param rule AclRule object from KafkaUser CR
      * @return The SimpleAclRule.
      */
-    public static SimpleAclRule fromCrd(AclRule rule)   {
-        return new SimpleAclRule(rule.getType(), SimpleAclRuleResource.fromCrd(rule.getResource()), rule.getHost(), rule.getOperation());
+    @SuppressWarnings("deprecation")
+    public static List<SimpleAclRule> fromCrd(AclRule rule) {
+        if (rule.getOperations() != null) {
+            List<SimpleAclRule> simpleAclRules = new ArrayList<>();
+            for (AclOperation operation : rule.getOperations()) {
+                simpleAclRules.add(new SimpleAclRule(rule.getType(), SimpleAclRuleResource.fromCrd(rule.getResource()), rule.getHost(), operation));
+            }
+            return simpleAclRules;
+        } else if (rule.getOperation() != null) {
+            return List.of(new SimpleAclRule(rule.getType(), SimpleAclRuleResource.fromCrd(rule.getResource()), rule.getHost(), rule.getOperation()));
+        } else {
+            throw new InvalidResourceException("Both fields `operations` and `operation` cannot be filled in at the same time");
```

Comment:
I do not think this will work => It they will be both set, it will fall into the first section and use the `operations` field.

Reply:
I also added a simple test about this behaviour.